src/main/kotlin/org/rust/ide/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -222,6 +223,17 @@ class RsExtractFunctionConfig private constructor(
 
     companion object {
         fun create(file: PsiFile, start: Int, end: Int): RsExtractFunctionConfig? {
+            doCreate(file, start, end)?.let { return it }
+
+            file.findElementAtIgnoreWhitespaceAfter(end - 1)?.let { lastElement ->
+                if (lastElement.elementType in listOf(RsElementTypes.COMMA, RsElementTypes.SEMICOLON)) {
+                    return create(file, start, lastElement.startOffset)
+                }
+            }
+            return null
+        }
+
+        fun doCreate(file: PsiFile, start: Int, end: Int): RsExtractFunctionConfig? {
```

Comment:
I'm not sure if the new function really needed here. Looks like the new logic is only needed when `findStatementsOrExprInRange` returns an empty list. So maybe something like this would work:

```Kotlin
fun create(file: PsiFile, start: Int, end: Int): RsExtractFunctionConfig? {
    val elements = findStatementsOrExprInRange(file, start, end).asList()
    if (elements.isEmpty()) {
        val lastElement = file.findElementAtIgnoreWhitespaceAfter(end - 1) ?: return null
        ...
    }
```

Reply:
Since we don't need recursion, looks like separate function is needed